90zfsbootmenu/module-setup.sh

```diff
@@ -153,7 +153,15 @@ install() {
   fi
 
   # Synchronize initramfs and system hostid
-  HOSTID="$( hostid )"
-  # shellcheck disable=SC2154
-  echo -ne "\\x${HOSTID:6:2}\\x${HOSTID:4:2}\\x${HOSTID:2:2}\\x${HOSTID:0:2}" > "${initdir}/etc/hostid"
+  if [ -e /etc/hostid ]; then
+    inst /etc/hostid
+    type mark_hostonly >/dev/null 2>&1 && mark_hostonly /etc/hostid
+  else
+    HOSTID="$( hostid )"
+    if [ "${HOSTID}" != "00000000" ]; then
+      # shellcheck disable=SC2154
+      echo -ne "\\x${HOSTID:6:2}\\x${HOSTID:4:2}\\x${HOSTID:2:2}\\x${HOSTID:0:2}" > "${initdir}/etc/hostid"
```

Comment:
I'd leave a comment that this breaks big endian usage.

Reply:
Done